SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Build/Build.cs

```diff
@@ -86,10 +82,55 @@ public static void Main(string[] args)
             var noCompile = args.Count(arg => arg.ToLowerInvariant() == "-nocompile") > 0;
             var additionalUATArgs = string.Join(" ", args.Skip(4).Where(arg => arg.ToLowerInvariant() != "-nocompile"));
 
-            var stagingDir = Path.GetFullPath(Path.Combine("../spatial", "build", "unreal"));
-            var outputDir = Path.GetFullPath(Path.Combine("../spatial", "build", "assembly", "worker"));
+            var stagingDir = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(projectFile), "../spatial", "build", "unreal"));
+            var outputDir = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(projectFile), "../spatial", "build", "assembly", "worker"));
             var baseGameName = Path.GetFileNameWithoutExtension(projectFile);
 
+            // Get UNREAL_HOME without environment variable.
+            Console.WriteLine("Finding Unreal Engine build.");
+            string uproject = File.ReadAllText(projectFile, Encoding.UTF8);
```

Comment:
Why can't `Setup.bat` call this (or a different project that shares this implementation), rather than duplicating all of the logic in two places?

Reply:
Two things, I want to not have the scripts be coupled together / rely on a certain project structure, with future work I hope to remove `Setup.bat` too. But also, I believe this being a C# file means we need to have compiled it to call it, the `Setup.bat` is looking for MSBuildTool used by Unreal specifically to compile our C# Build scripts. This would mean calling the C# wouldn't work. For the sake of discussion, one option could be to move the "Find Unreal Engine" functionality into a separate script that the C# project could call, but again this is coupling scripts again which I want to avoid at the moment.